qa/L0_metrics/test.sh

```diff
@@ -94,6 +94,33 @@ kill $SERVER_PID
 wait $SERVER_PID
 
 
+# Test metrics interval by querying host and checking energy
+METRICS_INTERVAL_SECS=0.5
+SERVER_ARGS="$SERVER_ARGS --metrics-interval-secs=${METRICS_INTERVAL_SECS}"
+run_server
+if [ "$SERVER_PID" == "0" ]; then
+  echo -e "\n***\n*** Failed to start $SERVER\n***"
+  cat $SERVER_LOG
+  exit 1
+fi
+
+num_iterations=10
+prev_energy=`curl -s localhost:8002/metrics | awk '/nv_energy_consumption{/ {print $2}'`
+for (( i = 0; i < $num_iterations; ++i )); do
+  sleep $METRICS_INTERVAL_SECS
+  current_energy=`curl -s localhost:8002/metrics | awk '/nv_energy_consumption{/ {print $2}'`
+  if [ $current_energy == $prev_energy ]; then
```

Comment:
This seems like a very unstable way to test this. If energy just happens to be the same across 2 samples then it will fail. Or what stops you from sening all 10 curls in < 0.5 seconds and then you will definitely get the same value.

Reply:
From the tests I've run, the energy seems to be be increasing pretty quickly, it would have to be a very low power GPU for it to fail. 

I didn't get the question regarding all 10 curls in < 0.5 seconds. I sleep the test thread for 0.5ms between curls.

Reply:
Sorry, I missed the sleep...
I guess this is ok.

Reply:
If we had timestamps, i think we could make this better, but I didn't see any.